File: CSC330/Homework/Python/recursion/iterative.py
# Iterative Method to Compute Factorial
import sys

def fact(val) : 
      if(val < 0) :
            return "--NO--\n"
      elif(val == 0) or (val == 1) :
            return 1
      else :
            factVal = 1
            for i in range(val, 1, -1) : 
                  factVal = factVal * i
            return factVal
            
# Take command line input, compute the factorial, print it out            
userInput = int(sys.argv[1])
print("Factorial of " + str(userInput) + " : " + str(fact(userInput)))

File: CSC330/Homework/perl/recursion/iterative.pl
#!/usr/bin/perl

sub fact {
 my $val = $_[0];
 if ( $val < 0 )
 {
   return "-- ERROR --\n";
 }
 elsif ( $val == 0 || $val == 1) {
    return 1
 }
 else {
    my $factval = 1;
    for ( my $i= $val; $i>0; $i-- ) {
       $factval = $factval * $i;
       }
    return $factval
   }
}

# Take the command line input, compute the factorial of it, and print
# it out.
print fact($ARGV[0]) . "\n";